react/src/components/FileLinkingComponent.tsx

```diff
@@ -47,11 +45,18 @@ const useAutocompleteStyles = makeStyles(() => ({
 /* A cell for linking files to a dataset. */
 const FileLinkingComponent: React.FC<{
     values: LinkedFile[];
-    options: UnlinkedFile[];
     onEdit: (newValue: UnlinkedFile[]) => void;
+    setFiles?: React.Dispatch<React.SetStateAction<UnlinkedFile[]>>;
     disabled?: boolean;
     disableTooltip?: boolean;
-}> = ({ values, options, onEdit, disabled, disableTooltip }) => {
+    inputValue?: string;
+    onInputChange?: (newInputvalue: string) => void;
+}> = ({ values, onEdit, disabled, disableTooltip, inputValue, onInputChange }) => {
+    const debouncedSearchQuery = useDebounce(inputValue || "None", 600);
+
+    const files = useUnlinkedFilesQuery(debouncedSearchQuery);
```

Comment:
So for debouncing here, you changed both the frontend and the backend so that the frontend will still send a request, just to `/api/unlinked/None`, which the backend will ignore?

Reply:
Yes, that's right.

Reply:
This is mainly because we can't use the custom hook inside a conditional statement on the frontend, so a request would still need to be sent, but the backend would simply ignore it.